lib/src/sep/0099/hotswap.dart

```diff
@@ -0,0 +1,193 @@
+    final receiveDestinationAssetOperation = PaymentOperationBuilder(
+      accountId,
+      toAsset,
+      fromAssetBalanceObject.balance, // Ensuring a 1:1 exchange
+    ).setSourceAccount(hotswapHandlerAccountId).build();
+
+    final untrustSourceAssetOperation = ChangeTrustOperationBuilder(
+      fromAsset,
+      '0',
+    ).build();
+
+    txBuilder
+        .addOperation(trustDestinationAssetOperation)
+        .addOperation(depositSourceAssetOperation)
+        .addOperation(receiveDestinationAssetOperation)
+        .addOperation(untrustSourceAssetOperation);
+
+    final transaction = txBuilder.build();
+
+    transaction.sign(signer, network);
+
+    var transactionXdr = transaction.toEnvelopeXdrBase64();
+    var hotswapUrl = Util.appendEndpointToUrl(
+      _serverAddress,
+      'hotswap',
+    ).replace(
+      queryParameters: {
+        'transaction_xdr': transactionXdr,
+      },
+    );
+
+    final response = await httpClient.post(hotswapUrl);
+
+    if (response.statusCode != 200) {
+      throw Exception(
+        'Failed to get signed transaction: ${response.statusCode}',
+      );
+    }
+
+    final signedTransactionResponseData =
+        jsonDecode(response.body) as Map<String, dynamic>;
+    transactionXdr = signedTransactionResponseData['signed_tx_xdr'] as String;
+
+    final signedTransaction = Transaction.fromV1EnvelopeXdr(
+      XdrTransactionEnvelope.fromEnvelopeXdrString(
+        transactionXdr,
+      ).v1!,
+    );
+    return signedTransaction;
+  }
+}
+
+class HotswapRoute extends Response {
+  final String toAddress;
+  final String fromAsset;
+  final String toAsset;
+  final double minimumAmount;
+
+  HotswapRoute({
+    required this.toAddress,
+    required this.fromAsset,
+    required this.toAsset,
+    required this.minimumAmount,
+  });
+
+  factory HotswapRoute.fromJson(Map<String, dynamic> json) {
+    return HotswapRoute(
+      toAddress:
+          (json['hotswap_address'] ?? json['receivables_address']) as String,
+      fromAsset: (json['from_asset'] ?? json['you_send_asset']) as String,
+      toAsset: (json['to_asset'] ?? json['we_send_asset']) as String,
+      minimumAmount: double.parse(
+        (json['min_amount'] ?? json['minimum_amount']) as String,
+      ),
+    );
+  }
+}
+
+class _HotswapInfoResponse {
+  final List<HotswapRoute> hotswapRoutes;
+
+  _HotswapInfoResponse({required this.hotswapRoutes});
+
+  factory _HotswapInfoResponse.fromJson(Map<String, dynamic> json) {
+    if (json.containsKey('hotswaps')) {
+      return _HotswapInfoResponse(
+        hotswapRoutes: (json['hotswaps'] as List)
+            .map(
+              (hotswap) => HotswapRoute.fromJson(
+                {
+                  'hotswaps': [hotswap]
+                },
+              ),
+            )
+            .toList(),
+      );
+    } else {
+      // Single hotswap case
+      return _HotswapInfoResponse(
+        hotswapRoutes: [
+          HotswapRoute.fromJson(json),
```

Comment:
I proposed a change in naming for the properties, so you can't use the same object for parsing.

Reply:
Done.